plugins/module_utils/rds.py

```diff
@@ -156,52 +200,39 @@ def call_method(client, module, method_name, parameters):
     changed = True
     if not module.check_mode:
         wait = module.params.get('wait')
-        # TODO: stabilize by adding get_rds_method_attribute(method_name).extra_retry_codes
+        rds_method_attributes = get_rds_method_attribute(method_name, module)
+        retry_codes = rds_method_attributes.retry_codes
+        waiter = rds_method_attributes.waiter
+        resource = rds_method_attributes.resource
         method = getattr(client, method_name)
         try:
-            if method_name in ['modify_db_instance', 'promote_read_replica']:
-                # check if instance is in an available state first, if possible
-                if wait:
-                    wait_for_status(client, module, module.params['db_instance_identifier'], 'modify_db_instance')
-                result = AWSRetry.jittered_backoff(catch_extra_error_codes=['InvalidDBInstanceState'])(method)(**parameters)
-            elif method_name == 'modify_db_cluster':
-                # check if cluster is in an available state first, if possible
-                if wait:
-                    wait_for_status(client, module, module.params['db_cluster_identifier'], method_name)
-                result = AWSRetry.jittered_backoff(catch_extra_error_codes=['InvalidDBClusterStateFault'])(method)(**parameters)
-            else:
-                result = AWSRetry.jittered_backoff()(method)(**parameters)
+            result = AWSRetry.jittered_backoff(catch_extra_error_codes=retry_codes)(method)(**parameters)
         except (BotoCoreError, ClientError) as e:
             changed = handle_errors(module, e, method_name, parameters)
 
         if wait and changed:
             identifier = get_final_identifier(method_name, module)
-            wait_for_status(client, module, identifier, method_name)
+            wait_for_status(client, module, resource, identifier, waiter)
```

Comment:
I guess you can directly pass `rds_method_attributes` as parameter instead of resource and waiter.

Reply:
@alinabuzachis I suppose - but then id have to unpack them right after anyway so isn't it more efficient to pass in the way it is now?

Reply:
@jatorcasso Mine was only a suggestion. In theory, those will be unpacked only when a waiter will be needed, otherwise, they won't (because useless). This was to also reduce the number of params passed.

Reply:
I was just curious haha. `wait_for_status` only gets called when wait is True, so we'll always need the waiter in that case

Reply:
Exactly, only in that case, but these are very small neats. However, I reviewed again and it looks good  for me.

Reply:
@jatorcasso Thank you.

Reply:
thanks!